parsec/core/fs/storage/workspace_storage.py

```diff
@@ -183,7 +183,10 @@ async def run(
 
     @manage_operational_error
     async def clear_memory_cache(self, flush: bool = True) -> None:
-        return await trio.to_thread.run_sync(self.sync_instance.clear_memory_cache, flush)
+        if flush:
+            return await trio.to_thread.run_sync(self.sync_instance.clear_memory_cache, flush)
+        else:
+            return self.sync_instance.clear_memory_cache(flush)
```

Comment:
```suggestion
            return await trio.to_thread.run_sync(self.sync_instance.clear_memory_cache, True)
        else:
            return self.sync_instance.clear_memory_cache(False)
```

Could be more explicit :thinking: